### PR TITLE
Refact/add http client

### PR DESCRIPTION
**Pull request by mardiros**
Add a way to configure the transport/client to publish trace.

Currently we are using 'scribe' and it stay the default.

I am adding an http client that use the `/v1/span` api and keep the `thrift` serialization.

Changing the content type to use protobuf for instance is more work and require to update the "model" part.

### CODE REVIEW

File: zipkin/transport/httpclient.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            cls.port = int(settings[prefix + 'collector.port'])
+
+        cls._url = '%s://%s:%s/api/v1/spans' % (
+            cls.scheme, cls.host, cls.port
+        )
+
+
+    @classmethod
+    def log(cls, trace):
+        if cls._url is None:
```
Comment (gdchamal) [inline zipkin/transport/httpclient.py:26]: the default value for the class will never permit to reach this condition

File: zipkin/transport/scribeclient.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                                 % (cls.host, cls.port))
+        return cls._client
+
+    @classmethod
+    def log(cls, trace):
+        if not cls.host:
+            logger.debug('Zipkin tracing is disabled')
+            return
+
+        logger.error('logging trace %s' % trace.trace_id)
```
Comment (gdchamal) [inline zipkin/transport/scribeclient.py:165]: error level ?

**Review by gdchamal — COMMENTED** [review]
Just some changes and it's ok for me

File: zipkin/util.py [hunk 3]
```diff
@@ -5,6 +5,8 @@
 from base64 import b64encode
 from six import text_type
 from thriftpy2.protocol import TBinaryProtocol
+from thriftpy2.thrift import TType
+from thriftpy2.protocol.binary import write_list_begin
 from thriftpy2.transport import TMemoryBuffer
 from thriftpy2.thrift import TDecodeException
```
Comment (Themimitoof) [inline zipkin/util.py:11]:
```suggestion
import random
import socket
import struct
from base64 import b64encode

from six import text_type
from thriftpy2.protocol import TBinaryProtocol
from thriftpy2.protocol.binary import write_list_begin
from thriftpy2.thrift import TDecodeException, TType
from thriftpy2.transport import TMemoryBuffer
```

File: zipkin/util.py [hunk 4]
```diff
@@ -111,3 +113,62 @@ def u64_as_i64(value):
         return data[0]
     except struct.error as e:
         raise ValueError(e)
+
```
Comment (Themimitoof) [inline zipkin/util.py:116]:
```suggestion
```

File: zipkin/transport/scribeclient.py (hunk 2 above)
Comment (Themimitoof) [reply zipkin/transport/scribeclient.py:165]:
```suggestion
        logger.info('logging trace %s' % trace.trace_id)
```

File: zipkin/transport/scribeclient.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+                    logger.error('Broken pipe while logging a trace '
+                                 'on zipkin collector %s:%d',
+                                 cls.host, cls.port)
+                else:
+                    logger.exception(unknown)
+            except Exception:
+                cls._client = None
+                logger.exception(unknown)
+        else:
+            logger.warn("Can't log zipkin trace, not connected")
```
Comment (Themimitoof) [inline zipkin/transport/scribeclient.py:199]: hasn't `logging.warn` been deprecated in python 3? :thinking: 

```suggestion
            logger.warning("Can't log zipkin trace, not connected")
```

**Review by Themimitoof — CHANGES_REQUESTED** [review]

File: zipkin/transport/scribeclient.py (hunk 5 above)
Comment (Themimitoof) [reply zipkin/transport/scribeclient.py:199]: Don't remember the version but @grigouze have made a PR about this in 2019: #4

File: zipkin/transport/httpclient.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    _url = 'http://zipkin:9411/api/v1/spans'
+
+    @classmethod
+    def configure(cls, settings, prefix):
+        cls.host = settings.get(prefix + 'collector')
+        if prefix + 'collector.port' in settings:
+            cls.port = int(settings[prefix + 'collector.port'])
+
+        cls._url = '%s://%s:%s/api/v1/spans' % (
+            cls.scheme, cls.host, cls.port
```
[the remark this answers is not in the document]
Comment (mardiros) [reply zipkin/transport/httpclient.py:20]: No, that just I've test without configuring it with default valuees...

File: zipkin/transport/scribeclient.py [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+from thriftpy2.transport import TTransportException, TFramedTransportFactory,\
+    TSocket
+from thriftpy2.protocol import TBinaryProtocolFactory
+from thriftpy2.protocol.binary import write_message_begin, write_val
+from thriftpy2.thrift import TClient, TType, TMessageType
+
+from ..util import base64_thrift_formatter
+from .scribe import scribe_thrift
+from ..client import Local
```
[the remark this answers is not in the document]
Comment (mardiros) [reply zipkin/transport/scribeclient.py]: DONE and fixup commit

File: zipkin/transport/scribeclient.py (hunk 5 above)
Comment (mardiros) [reply zipkin/transport/scribeclient.py:217]: Actually i've move the file client.py to transport/scribeclient.py without rewriting the content...

**Pull request merged**